test_results.py

```diff
@@ -0,0 +1,172 @@
+
+        Returns:
+            Pass: If the scode's center is not same as cscode's center
+            Fail: If the scode's center is same as cscode's center
+        """
+        scf = ParseTSVFile(self.school_center_file)
+        data = scf.get_rows()
+        scodes_centers = []
+
+        scodes_centers = get_scode_cscode_id(data)
+
+        # Check if there are any duplicates id if duplicate then there is a collision between school and center
+        duplicates = [
+            item for item in set(scodes_centers) if scodes_centers.count(item) > 1
+        ]
+
+        self.assertFalse(
+            duplicates,
+            f"Duplicate values found in scode_center_code: {', '.join(duplicates)}",
+        )
+
+    def test_center_schools_same_management(self):
+        """_Test if the schools under same management do not have the coinciding center_
+          Test case ID :- एकै स्वामित्व / व्यवस्थापनको भनी पहिचान भएका केन्द्रमा पार्न नहुने
+
+        Returns:
+            Pass: If the schools with same management are not each other's center
+            Fail: If the schools with same management are each other's center
+        """
+        scf = ParseTSVFile(self.school_center_file)
+        cpf = ParseTSVFile(self.school_center_pref_file)
+        data_scf = scf.get_rows()
+        data_cpf = cpf.get_rows()
+        for cpf_data in data_cpf:
+            if "same_management" not in cpf_data:
+                data_cpf.remove(cpf_data)
+
+        failures = []
+
+        scodes_centers = get_scode_cscode_id(data_scf)
+
+        same_management_scodes_centers = get_scode_cscode_id(data_cpf)
+        for same_management_center in same_management_scodes_centers:
+            try:
+                assert (
+                    same_management_center not in scodes_centers
+                ), f"Schools with same management have same center {same_management_center}"
+            except AssertionError:
+                failures.append(
+                    f"Schools with same management have same center {same_management_center}"
+                )
+        if failures:
+            raise AssertionError(failures)
+
+    def test_center_schools_problematic_history(self):
+        """_Test if the schools with problemetic history do not have the coinciding center_
+          Test case ID :- विगतमा कुनै विद्यालयको कुनै केन्द्रमा पार्दा समस्या देखिएकोमा केन्द्र नदोहोऱ्याउन नहुने
+
+        Returns:
+            Pass: If the schools with problemetic history are not each other's center
+            Fail: If the schools with problemetic history are each other's center
+        """
+        scf = ParseTSVFile(self.school_center_file)
+        cpf = ParseTSVFile(self.school_center_pref_file)
+        data_scf = scf.get_rows()
+        data_cpf = cpf.get_rows()
+        for cpf_data in data_cpf:
+            if "problematic history" not in cpf_data:
```

Comment:
instead of checking for exact match in remarks field which is probably a open text field, it might better to merge this test_center_school_same_management and just check if cscode, scode pair has pref score less than PREF_CUTOFF

Reply:
Done